### PR TITLE
Create rule S6288[java]: Authorizing non-authenticated users to use keys in the Android KeyStore is security-sensitive

### PR DESCRIPTION
**Pull request by eric-therond-sonarsource**
https://jira.sonarsource.com/browse/SONARJAVA-3868

### CODE REVIEW

File: rules/S6288/java/rule.adoc [hunk 1]
```diff
@@ -1 +1,39 @@
-include::../rule.adoc[]
+include::../description.adoc[]
+
+include::../ask-yourself.adoc[]
+
+include::../recommended.adoc[]
+
+== Noncompliant Code Example
+
+Any users can use the key:
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6288/java/rule.adoc:9]: I think it should be `Any user` instead.

File: rules/S6288/java/rule.adoc [hunk 2]
```diff
@@ -1 +1,39 @@
...
+include::../recommended.adoc[]
+
+== Noncompliant Code Example
+
+Any users can use the key: 
+
+----
+KeyGenerator keyGenerator = KeyGenerator.getInstance(KeyProperties.KEY_ALGORITHM_AES, "AndroidKeyStore");
+
+KeyGenParameterSpec builder = new KeyGenParameterSpec.Builder("test_secret_key_noncompliant",KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT) // Noncompliant (s6288)
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6288/java/rule.adoc:14]: `(s6288)` probably not necessary in this code sample here.

File: rules/S6288/java/rule.adoc [hunk 3]
```diff
@@ -1 +1,39 @@
...
+
+The use of the key is limited to authenticated users (for a duration of time defined to 60 seconds):
+
+----
+KeyGenerator keyGenerator = KeyGenerator.getInstance(KeyProperties.KEY_ALGORITHM_AES, "AndroidKeyStore");
+
+KeyGenParameterSpec builder = new KeyGenParameterSpec.Builder("test_secret_key", KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT)
+    .setBlockModes(KeyProperties.BLOCK_MODE_GCM)
+    .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
+    .setUserAuthenticationRequired(true) // Compliant
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6288/java/rule.adoc:32]: Ann once told me we should not add the `Compliant` comments in the code samples here. Looking at rules.sonarsource.com that is not something we always followed though.

**Review by hendrik-buchwald-sonarsource — APPROVED** [review]
LGTM except for some small comments that I added.

File: rules/S6288/java/rule.adoc (hunk 3 above)
Comment (eric-therond-sonarsource) [reply rules/S6288/java/rule.adoc:32]: I just saw your comment right now :p
Alex found the same problem than you (followed what was presented this morning)

**Pull request merged**